grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/fragment/OutSubFragment.java

```diff
@@ -53,7 +58,7 @@ public String name() {
 
     @Override
     public double fragmentCost() {
-        return COST_SAME_AS_PREVIOUS;
+        return accurateFragmentCost.orElse(COST_SAME_AS_PREVIOUS);
```

Comment:
This "default" behaviour using `accurateFragmentCost` needs to move down to the `Fragment` interface.

I'd suggest having something like:

```
default double fragmentCost() {
    return accurateFragmentCost.orElse(internalFragmentCost());
}

double internalFragmentCost();
```

Then everything implements `internalFragmentCost()` instead of `fragmentCost()`.